### ISSUE TITLE
Small improvements to make 

### ISSUE THREAD

**Issue by rosikand** [rosikand/torchplate#10]
In this thread, I'll enumerate some smaller changes to make. 

- Save `epoch_num` as a global variable in `train` such that child classes can access epoch state without needing to keep track of the state themselves.

**Comment by rosikand** [comment]
- Support multi-node training with Hugging Face Accelerate

**Comment by rosikand** [comment]
- Gradient accumulation option in `train`

**Comment by rosikand** [comment]
- save model weights every `n` epochs option

**Comment by rosikand** [comment]
print loss at each step with `pbar`. See [here](https://github.com/lucidrains/denoising-diffusion-pytorch/blob/main/denoising_diffusion_pytorch/denoising_diffusion_pytorch.py#L902).

**Comment by rosikand** [comment]
- Automatic `torch.device` placement

**Comment by rosikand** [comment]
Nearly every feature mentioned thus far has been added in `version 0.0.7`.
